Branch-SDK/Branch-SDK/BNCServerInterface.m

```diff
@@ -239,26 +176,21 @@ - (NSURLRequest *)preparePostRequest:(NSDictionary *)post url:(NSString *)url re
     return request;
 }
 
-- (BNCServerResponse *)processServerResponse:(NSURLResponse *)response data:(NSData *)POSTReply error:(NSError *)error tag:(NSString *)requestTag andLinkData:(BNCLinkData *)linkData {
+- (BNCServerResponse *)processServerResponse:(NSURLResponse *)response data:(NSData *)data error:(NSError *)error tag:(NSString *)requestTag andLinkData:(BNCLinkData *)linkData {
     BNCServerResponse *serverResponse = nil;
     if (!error) {
         NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
-        NSNumber *statusCode = [NSNumber numberWithLong:[httpResponse statusCode]];
+        NSNumber *statusCode = @([httpResponse statusCode]);
         
         serverResponse = [[BNCServerResponse alloc] initWithTag:requestTag andStatusCode:statusCode];
         serverResponse.linkData = linkData;
-        
-        if (POSTReply != nil) {
-            NSError *convError;
-            id jsonData = [NSJSONSerialization JSONObjectWithData:POSTReply options:NSJSONReadingMutableContainers error:&convError];
-            if (!convError) {
-                serverResponse.data = jsonData;
-            }
-        }
-    } else {
-        serverResponse = [[BNCServerResponse alloc] initWithTag:requestTag andStatusCode:[NSNumber numberWithInteger:error.code]];
+        serverResponse.data = [BNCEncodingUtils decodeJsonDataToDictionary:data];
+    }
+    else {
+        serverResponse = [[BNCServerResponse alloc] initWithTag:requestTag andStatusCode:@(error.code)];
```

Comment:
minor, but since we create this twice, maybe have this outside the if / else block, and have a setter/getter for the status code accordingly?